crates/storage/src/snapshot.rs

```diff
@@ -43,18 +45,30 @@ impl std::fmt::Debug for Snapshot {
 // We don't want to expose the `TreeReader` implementation outside of this crate.
 #[derive(Debug)]
 pub(crate) struct Inner {
-    snapshot: RocksDbSnapshot,
-    version: jmt::Version,
+    pub(crate) multistore: Multistore,
+    pub(crate) snapshot: RocksDbSnapshot,
+    pub(crate) version: jmt::Version,
     // Used to retrieve column family handles.
-    db: Arc<rocksdb::DB>,
+    pub(crate) db: Arc<rocksdb::DB>,
 }
 
 impl Snapshot {
-    pub(crate) fn new(db: Arc<rocksdb::DB>, version: jmt::Version) -> Self {
+    /// Creates a new `Snapshot` with the given version and substore configs.
+    pub(crate) fn new(
+        db: Arc<rocksdb::DB>,
+        version: jmt::Version,
+        mut substore_configs: Vec<Arc<store::substore::SubstoreConfig>>,
+    ) -> Self {
+        if substore_configs.is_empty() {
+            let transparent_config = store::substore::SubstoreConfig::transparent_store();
+            substore_configs.push(Arc::new(transparent_config));
+        }
+
```

Comment:
I could be missing something but this doesn't seem right to me. Why are we creating a new config object here?